source/_posts/2016/2016-12-24-how-to-avoid-inject-thanks-to-decorator-feature-in-nette.md

```diff
@@ -0,0 +1,254 @@
+---
+layout: post
+title: "How to avoid @inject thanks to Decorator feature in Nette"
+perex: "I often find <code>@inject</code> being overused in projects I review while mentoring. They often bring less writing, but in exchange they break SOLID principals. Today I will show you solution that will <strong>keep your code both small and clean</strong> - <strong>Decorator feature in Nette</strong>."
+lang: en
+---
+
+As [Derek Simons says](https://www.ted.com/talks/simon_sinek_how_great_leaders_inspire_action) says...
+
+## ...Start with "Why"
+
+Why am I writing this article? I try to improve knowledge interoperability between frameworks by finding the common lowest denominator, so it **is easier to understand and use each other**.
+
+I understand every framework has its own specific way how to solve same issues, but it creates *knowledge vendor lock*. So even if I know how to solve some problem in Nette, I cannot do that in Symfony and have to re-learn it again.
+ 
+Today, I will try to agree on setter injection with you. 
```

Comment:
Meaning set* methods + some service setup decorator implementation? This article actually feels a lot nette-specific since it mentions inject annotations several times. Or better said moving from nette-specific things to something that may be common for multiple frameworks - that lowest denominator you mentioned is apparently set* methods solution in this case? If you rewrite it to say that the goal is to discourage nette- (or framework-) specific things in favor of those that may be common it might be better.